pkg/run/run.go

```diff
@@ -0,0 +1,160 @@
+// Copyright 2024 Outreach Corporation. All Rights Reserved.
+
+// Description: Entrypoint to run the app
+
+// Package run provides a function that can be invoked inside of main to set up
+// all the service-standard components we expect every service to run.
+//
+// clients should provide any runners they require as part of the app to the
+// Run function with OptAddRunner, and then call `Run` in their main function.
+package run
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"net/http"
+	"os"
+
+	"github.com/getoutreach/gobox/pkg/app"
+	"github.com/getoutreach/gobox/pkg/async"
+	"github.com/getoutreach/gobox/pkg/env"
+	"github.com/getoutreach/gobox/pkg/olog"
+	"github.com/getoutreach/gobox/pkg/trace"
+	"github.com/getoutreach/stencil-golang/pkg/serviceactivities/automemlimit"
+	"github.com/getoutreach/stencil-golang/pkg/serviceactivities/gomaxprocs"
+	"github.com/getoutreach/stencil-golang/pkg/serviceactivities/shutdown"
+
+	"github.com/getoutreach/httpx/pkg/handlers"
+)
+
+// runOpts are the options to pass to Run to configure your service
+type runOpts struct {
+	httpAppHandler http.Handler
+	log            *slog.Logger
+	httpAddr       string
+	runners        []async.Runner
+}
+
+// Option is the interface for an  option
+type Option interface {
+	apply(*runOpts) error
+}
+
+// optionFunc is a function that implements serviceOption
+type optionFunc func(*runOpts) error
+
+// apply implements serviceOption.
+func (s optionFunc) apply(o *runOpts) error {
+	return s(o)
+}
+
+// WithLogger sets the logger. Otherwise it defaults to olog.New
+func WithLogger(l *slog.Logger) Option {
+	return optionFunc(func(o *runOpts) error {
+		o.log = l
+		return nil
+	})
+}
```

Comment:
thought: I wonder if this is the right way to do this, given that `olog`-based logging seems to be package-based now (as opposed to `gobox/pkg/log` which is effectively a global)

Reply:
My thought was that this is more flexible for teams to configure the logger, since some configuration options aren't super accessible yet